autoload/maktaba/buffer.vim

```diff
@@ -10,6 +10,29 @@ endfunction
 
 
 ""
+" Gets the text of the current or last visual selection.
+" Useful for visual mode mappings.
+function! maktaba#buffer#GetVisualSelection() abort
+  let [l:lnum1, l:col1] = getpos("'<")[1:2]
+  let [l:lnum2, l:col2] = getpos("'>")[1:2]
+  " 'selection' is a rarely-used option for overriding whether the last
+  " character is included in the selection. Bizarrely, it always affects the
+  " last character even when selecting from the end backwards.
+  if &selection !=# 'inclusive'
```

Comment:
Oh wow, I'd never heard of that setting. I added a case covering it.

I did discover in the process that this is _not_ the same behavior you get if you yank odd visual selections with `virtualedit=all` active. Yank will pad with spaces if you select part of a tab or past the end of a line. I believe our behavior is more intuitive for use in visual mode mappings and scripts.